src/test_l1_dsAudio.c

```diff
@@ -99,6 +100,92 @@ intptr_t null_handle = 0;
 
 /* Global flags to support features */
 static bool extendedEnumsSupported=false; //Default to not supported
+static char* device_type = NULL;
+static char* device_name = NULL;
+static uint16_t numports = 0;
+ 
+typedef struct  port_config {
+	uint16_t typeid;
+	uint16_t index;
+	char* port_name;
+	int no_of_supported_compression;
+	int supported_compressions[ARRAY_SIZE];
+	int no_of_supported_encodings;
+	int supported_encodings[ARRAY_SIZE];
+	int no_of_supported_stereo_mode;
+	int supported_stereo_mode[ARRAY_SIZE];
+	int ms12_capabilites;
+	char* ms12_audio_profiles[ARRAY_SIZE];
+	int ms12_audioprofilecount;
+	bool isms12decode;
+	bool isms11decode;
+	int atmos_capabilites;	 
+
+}port_config_t;
+
+port_config_t* portid = NULL;
+
+void populate_port_config()
+{
+	printf("entered in to the populate_port_config \n"); 
+	ut_kvp_status_t status;
+
+	status = ut_kvp_getStringField(ut_kvp_profile_getInstance(), "Device/Type", device_type, UT_KVP_MAX_ELEMENT_SIZE);
+   	status = ut_kvp_getStringField(ut_kvp_profile_getInstance(), "Device/Name", device_name, UT_KVP_MAX_ELEMENT_SIZE);
+	numports = ut_kvp_getUInt16Field( ut_kvp_profile_getInstance(), "Number_of_supported_ports" );
+
+	portid = (port_config_t*) malloc(numports * sizeof(port_config_t));
+	if(portid == NULL)
+	{
+		UT_LOG("\n %s - Failed to allocate memory \n",__FUNCTION__);
+	}
+
+	for(int i =0 ; i < numports ; i++)
+	{
+		portid[i].typeid = ut_kvp_getUInt32Field(ut_kvp_profile_getInstance(), "ports/i/Typeid");
+		portid[i].index = ut_kvp_getUInt32Field(ut_kvp_profile_getInstance(), "ports/i/connected_video_ports/index");
```

Comment:
"i" inside strong will not be replaced , 
we need tomise sscanf or similar method to replace the value in strong before calling kvp apis

Reply:
done , yes it wont be replace.